### DISCUSSION TITLE
Checkerboard Artifacts

### DISCUSSION THREAD

**Discussion by zachcoleman** [developmentseed/titiler#721] [Q&A] [answered]
Does anyone have any insight into why checkerboard artifacts like this may occur when zooming in?

I am using TiTiler against a COG with no levels.

Zoomed out I have:
![Screenshot 2023-11-01 at 11 33 05 AM](https://github.com/developmentseed/titiler/assets/42484306/28a4e99f-7482-4f0e-9814-b62d5919fa4f)
 
 As I zoom in however I get:
![Screenshot 2023-11-01 at 11 33 19 AM](https://github.com/developmentseed/titiler/assets/42484306/a2b276f1-d794-4d3b-8f3a-b1576d871a30)

Here's an example of one of the tiles produced:
![Screenshot 2023-11-01 at 11 34 45 AM](https://github.com/developmentseed/titiler/assets/42484306/9fc62562-977f-47e2-acf7-544a053aed39)

Any help understanding what's going would be appreciated.

**Comment by vincentsarago** [comment]
@zachcoleman can you share the data? 

Can you try using `resampling=cubic`

**Reply by zachcoleman** [reply]
@vincentsarago Thank you! That definitely makes the individual tiles look smoother when the sub-pixel calculations are happening vs the nearest-neighbor approach that gives the hard lines. 


However, I'm curious why are the tiles still repeated when sub-pixel rendering is occurring: 
![Screenshot 2023-11-02 at 9 34 56 AM](https://github.com/developmentseed/titiler/assets/42484306/0f911d2c-1eed-4d6b-9643-167775e24193)

I apologize, but I will have to try and get approval to send the data and get back to you on that.

**Reply by vincentsarago** [reply]
could you run `gdalinfo` and post the result here?

**Reply by zachcoleman** [reply]
@vincentsarago I think I understand what's going on. 

I believe this from each tile's values are calculated relatively and individually with respect to that tile's pixel coordinates vs more absolutely with respect to all the tiles being generated in a space. There's mostly likely not an easy solution to this. 

![image](https://github.com/developmentseed/titiler/assets/42484306/c03b5a9b-7f90-4bd8-98fa-f535990e5084)

**Reply by vincentsarago** [reply, marked as answer]
I guess you are trying to `overzoom` (creating tiles at higher zoom level than the resolution of your data) meaning that you are telling the tiler to create data for a full tile, using maybe one or two pixel of your data. 🤷

**Reply by zachcoleman** [reply]
@vincentsarago Yes we are over-zooming and I agree the sub-pixel and values are having to be interpolated from the four pixel values surrounding it. 

I think this comes from each tile being calculated independently and interpolation (regardless of method) is per a tile within each individual tile's region.

**Reply by vincentsarago** [reply]
> I think this comes from each tile being calculated independently and interpolation (regardless of method) is per a tile within the sub-pixel region.

Yes, you can try to add `padding=4` to see if it returns better images but I'm not sure it will change much

**Reply by zachcoleman** [reply]
@vincentsarago The odd part to me is that the interpolation seems to be aware of where the tile is located within the entire sub-pixel region, but interpolation is still only within the tile. I'm not sure entirely sure. I made a demo of what we are seeing. 

The nearest here seems to be scaled within a tile somehow like:
![image](https://github.com/developmentseed/titiler/assets/42484306/fdd2e1e1-a9f3-4a19-a1eb-68f91210ab6e)

The tile seems to be knowledgeable of where it is relatively in the sub-pixel region that creates the growing and shrinking color regions. Also seen in the previous example, but this relative calculation seems to affect distances to the tile corners and not the absolute distances to source pixel locations. 

In reality the nearest values if calculated absolutely here would be like this instead:
![Screenshot 2023-11-02 at 10 29 18 AM](https://github.com/developmentseed/titiler/assets/42484306/e85e3f73-a192-4057-906b-56f8f43511fb)

**Reply by zachcoleman** [reply]
Thanks for help @vincentsarago
